crates/solver/src/settlement_ranker.rs

```diff
@@ -164,11 +164,33 @@ impl SettlementRanker {
             );
         }
 
-        let (mut rated_settlements, errors) = self
+        let (rated_settlements, errors) = self
             .settlement_rater
             .rate_settlements(solver_settlements, external_prices, gas_price)
             .await?;
 
+        // Discard solutions with negative obj value
+        let mut rated_settlements = rated_settlements
+            .into_iter()
+            .filter(|(solver, rated_settlement, _)| {
+                // TODO for now, apply only for 0x solver
+                if solver.name() != "0x" {
+                    return true;
+                }
+
+                let objective_value = rated_settlement.objective_value();
+                let objective_value_positive = objective_value.ge(&Ratio::from_integer(0.into()));
```

Comment:
This could be slightly easier to read with `Ratio::zero()`.

```suggestion
                let objective_value_positive = objective_value.ge(&Ratio::zero());
```

Reply:
Nice.